### ISSUE TITLE
Cluster Subnet CIDR only defined for us-south zones 1-2-3

### ISSUE THREAD

**Issue by srcarrier** [kubernetes-sigs/cluster-api-provider-ibmcloud#319]
/kind bug
/area provider/ibmcloud

**What steps did you take and what happened:**
CIDR values for subnet creation are hard-coded for us-south zones 1-2-3 only, thus if a user attempts to use a different region (e.g. `us-east`), subnet provisioning fails.

https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/ff3075371756d25c47fd640789bc4db1b9895687/cloud/scope/cluster.go#L198-L205

**What did you expect to happen:**

I expect the ability to define any region/zone supported by the target workload cluster environment (IBM Cloud VPC2).

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version: v1alpha3
- Minikube/KIND version: kind v0.10.0 go1.15.7 darwin/amd64
- Kubernetes version: (use `kubectl version`):
```Client Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-beta.0", GitCommit:"14ede42c4fe699a7078b566d89abc160f26857a2", GitTreeState:"clean", BuildDate:"2020-01-15T11:15:40Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-21T01:11:42Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`): MacOS 11.2.3

**Comment by mkumatag** [comment]
@emmayang any new updates on this? are you still working on this issue?

**Comment by mkumatag** [comment]
I see an API here for getting the address prefix for all the zones - https://cloud.ibm.com/apidocs/vpc#list-vpc-address-prefixes, can be used to filter the subnet based on the zone mentioned.

**Comment by emmayang** [comment]
Yes, this API would do the work, will add this one call before creating subnet, and then use the retrieved CIDR prefix. 

So sorry for the late response, met some problems with my dev environment, and working on fixing it...

**Comment by mkumatag** [comment]
> The VPC SDK is still using the old response schema, while staging already switched to latest schema, and introduced a breaking change. And should be noted, that the VPC provider is broken when API_Endpoint is staging.

Is something got deprecated? Please add the TODO so that we fix that when it rolled out to the production.

**Comment by emmayang** [comment]
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/345 has been opened to track the version change issue.
